scout/commands/base.py

```diff
@@ -28,97 +24,46 @@
 from scout.commands.serve import serve
 from scout.commands.update import update as update_command
 from scout.commands.index_command import index as index_command
-
-from scout.adapter.utils import check_connection
-
-try:
-    from scoutconfig import *
-except ImportError:
-    pass
+from scout.server import extensions
 
 LOG_LEVELS = ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL']
 LOG = logging.getLogger(__name__)
 
 
-@click.group()
+@click.group(cls=FlaskGroup, create_app=create_app)
 @click.option('--loglevel', default='INFO', type=click.Choice(LOG_LEVELS),
               help="Set the level of log output.", show_default=True)
-@click.option('-db', '--mongodb', help='Name of mongo database [scout]')
-@click.option('-u', '--username')
-@click.option('-p', '--password')
-@click.option('-a', '--authdb', help='database to use for authentication')
-@click.option('-port', '--port', help="Specify on what port to listen for the mongod")
-@click.option('-h', '--host', help="Specify the host for the mongo database.")
-@click.option('-c', '--config', type=click.Path(exists=True),
-              help="Specify the path to a config file with database info.")
 @click.option('--demo', is_flag=True, help="If the demo database should be used")
-@click.version_option(__version__)
-@click.pass_context
-def cli(context, mongodb, username, password, authdb, host, port, loglevel, config, demo):
-    """scout: manage interactions with a scout instance."""
-    # log_format = "%(message)s" if sys.stdout.isatty() else None
+@click.option('-c', '--config', type=click.Path(exists=True))
+@with_appcontext
+def app_cli(loglevel, demo, config):
+    """Entry point of Scout CLI"""
```

Comment:
I don't think I would have understood either version when I ran "scout" for the first time. How about at least merging them:

```suggestion
    """Scout CLI:  manage interactions with a scout instance."""
```
Ah, and the corresponding test will need a change then! :blush: